tests/escrow_calls.py

```diff
@@ -0,0 +1,82 @@
+
+
+from eip712_structs import make_domain
+from web3 import Web3
+from web3.exceptions import ContractCustomError, ContractLogicError
+import json
+from eth_account.messages import encode_defunct
+import time
+from helpers import ReceiptAggregateVoucher, decode_custom_error, obtain_final_signature
+from helpers import GATEWAY, ALLOCATION_ID, ALLOCATIONID_PK, SIGNER, SIGNER_PK, RECEIVER, ESCROW_ADDRESS, TAP_ADDRESS
+
+# This script will help test that the subgraph is actually catching the required information
+
+
+my_domain = make_domain(name='tapVerifier',
+    version='1.0',
+    chainId=1337,
+    verifyingContract=TAP_ADDRESS
+)
+
+
+w3 = Web3(Web3.HTTPProvider('http://127.0.0.1:8545'))
+
+#RAV
+msg = ReceiptAggregateVoucher()
+msg['allocationId'] = ALLOCATION_ID
+msg['timestampNs'] = 1691694025
+msg['valueAggregate'] = 5
+
+#EIP712
+signable_bytes = msg.signable_bytes(my_domain)
+rav_digest = encode_defunct(signable_bytes)
+rav_signature = obtain_final_signature(SIGNER_PK, signable_bytes)
```

Comment:
This could be greatly simplified with https://github.com/ApeWorX/eip712:
```py
from eip712.messages import EIP712Message

from web3 import Web3
from eth_account.account import Account

class RAV(EIP712Message):
    _chainId_ = 1337
    _verifyingContract_ = TAP_ADDRESS
    _version_ = "1.0"

    allocationId: 'address'
    timestampNs: 'uint256'
    valueAggregate: 'uint256'

msg = RAV(allocationId=ALLOCATION_ID, timestampNs=1691694025, valueAggregate=5)

#EIP712
account = Account.from_key(SIGNER_PK)
rav_signature = account.sign_message(msg.signable_message)
```

Also allows ditching the `sha3` module that is so old that I can't install it on my machine :frowning_face:, as well as `coincurve`

Reply:
np, I can try this once more using this method, that way is how I got it working in the end, but yeah now that it works I can go back and look into that module